routing/state.go

```diff
@@ -0,0 +1,157 @@
+package routing
+
+import (
+	"errors"
+	"github.com/gogo/protobuf/proto"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pb "github.com/libp2p/go-libp2p-core/routing/pb"
+	ma "github.com/multiformats/go-multiaddr"
+	"time"
+)
+
+// The domain string used for routing state records contained in a SignedEnvelope.
+const StateEnvelopeDomain = "libp2p-routing-state"
```

Comment:
I think routing is a bad name for this. Routing is _a process_ for which a record of this kind _can_ be used. But it's not the only one. I'd rather call this a "signed peer record", and put it in the peer package.

Reply:
Sounds good to me 👍